### PR TITLE
misc: Various fixes for makedeps script

### PR DESCRIPTION
**Pull request by ofrzeta**
- set GOBIN to GOPATH/bin if not set

- install Go distribution package only if no go binary found

- use new govet home to import govet

## Tips:

* please read the style guide before submitting your patch:
[docs/style-guide.md](../docs/style-guide.md)

* commit message titles must be in the form:
```topic: Capitalized message with no trailing period```
or:
```topic, topic2: Capitalized message with no trailing period```

* golang code must be formatted according to the standard, please run:
```
make gofmt		# formats the entire project correctly
```
or format a single golang file correctly:
```
gofmt -w yourcode.go
```

* please rebase your patch against current git master:
```
git checkout master
git pull origin master
git checkout your-feature
git rebase master
git push your-remote your-feature
hub pull-request	# or submit with the github web ui
```

* after a patch review, please ping @purpleidea so we know to re-review:
```
# make changes based on reviews...
git add -p		# add new changes
git commit --amend	# combine with existing commit
git push your-remote your-feature -f
# now ping @purpleidea in the github PR since it doesn't notify us automatically
```

### CODE REVIEW

File: misc/make-deps.sh [hunk 1]
```diff
@@ -55,16 +59,23 @@ fi
 
 if [ $travis -eq 0 ]; then
 	if [ ! -z "$YUM" ]; then
+
+	    if [ -z "$GO" ]; then
```
Comment (purpleidea) [inline misc/make-deps.sh]: i think you've got a mix of tabs and spaces here and elsewhere. use only tabs please :)

File: misc/make-deps.sh [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 # setup a simple go environment
+if [[ -z "${GOBIN}" ]]; then
```
Comment (purpleidea) [inline misc/make-deps.sh]: This stuff should be in misc/make-path.sh ... can you see if that needs improving?

Comment (ofrzeta) [reply misc/make-deps.sh]: Sorry, I didn't notice. I removed those lines now.

**Comment by purpleidea** [comment]
@ofrzeta Can you rebase your patch into a single commit please?

**Comment by ofrzeta** [comment]
Like that?

**Comment by ofrzeta** [comment]
I guess I am :) I can't find anything about squashing serveral commits into one but that's what I did anyway I think. If you look at my latest (well, next-to-last) commit it contains all the changes (adding and removing stuff) folded into one, as far as I can see. looks like I only need to get rid of the other ones. And then there's always that master merge to update my local repo.

**Comment by purpleidea** [comment]
@ofrzeta At this point I'd recommend:

```
git checkout master
git pull
git checkout -b feat/my-new-branch
# copy your changes in
git add -p # until they're perfect
git commit # make a commit
git push yourremote feat/my-new-branch
```

HTH, I wish we had a better contributing guide at the moment, but there also some git tutorials online you should explore! HTH

**Comment by ofrzeta** [comment]
I sure read Git docs about rebasing and squashing commits and I guess I could create a much better PR if I started over but I don't get how I can do what you suggest ("copy your changes in") because my repository already contains all changes.

**Comment by purpleidea** [comment]
Closing in favour of #351

**Pull request closed without merging**